osu.Game/Overlays/Dialog/PopupDialog.cs

```diff
@@ -225,7 +225,12 @@ protected PopupDialog()
         /// <summary>
         /// Programmatically clicks the first button of the provided type.
         /// </summary>
-        public void PerformAction<T>() where T : PopupDialogButton => Buttons.OfType<T>().First().TriggerClick();
+        public void PerformAction<T>() where T : PopupDialogButton
+        {
+            // Buttons are regularly added in BDL or LoadComplete, so let's schedule to ensure
+            // they are ready to be pressed.
```

Comment:
As far as I can tell 99% usages actually add the buttons in _ctor_, and the crashing case is the _only_ one that does it in BDL. So it's a bit weird that this says this. But I agree that BDL should definitely be supported.